ion_schema_2_0/constraints/timestamp_offset.isl

```diff
@@ -0,0 +1,121 @@
+$ion_schema_2_0
+
+type::{
+  name: 'timestamp_offset: ["+02:43"]',
+  timestamp_offset: ["+02:43"],
+}
+
+$test::{
+  type: 'timestamp_offset: ["+02:43"]',
+  should_accept_as_valid:[
+    2000-01-01T00:00+02:43,
+    2000-01-01T00:00:00+02:43,
+    2000-01-01T00:00:00.000+02:43,
+  ],
+  should_reject_as_invalid:[
+    null,
+    null.timestamp,
+    "+2:43",
+    2000-01-01T00:00+03:43,
+    2000-01-01T00:00+02:44,
+    2000-01-01T00:00-02:43,
+    2000-01-01T,
+  ]
+}
+
+type::{
+  name: 'timestamp_offset: ["-00:00"]',
+  timestamp_offset: ["-00:00"],
+}
+
+$test::{
+  type: 'timestamp_offset: ["-00:00"]',
+  should_accept_as_valid:[
+    2000T,
+    2001-02-03T,
+    2001-02-03T04:05:06-00:00,
+  ],
+  should_reject_as_invalid:[
+    null,
+    null.timestamp,
+    2000-01-01T00:00+03:00,
+    2000-01-01T00:00+00:00,
+    2000-01-01T00:00-01:00,
+  ]
+}
+
+type::{
+  name: 'timestamp_offset: ["+00:00"]',
+  timestamp_offset: ["+00:00"],
+}
+
+$test::{
+  type: 'timestamp_offset: ["+00:00"]',
+  should_accept_as_valid:[
+    2000-01-01T00:00+00:00,
+    2000-01-01T12:34:56.789Z,
+  ],
+  should_reject_as_invalid:[
+    null,
+    null.timestamp,
+    2000-01-01T00:00+03:00,
+    2000-01-01T00:00-01:00,
+    2000T,
+    2001-02-03T,
+    2001-02-03T04:05:06-00:00,
+  ]
+}
+
+type::{
+  name: 'timestamp_offset: ["+01:23", "-02:34"]',
+  timestamp_offset: ["+01:23", "-02:34"],
+}
+
+$test::{
+  type: 'timestamp_offset: ["+01:23", "-02:34"]',
+  should_accept_as_valid:[
+    2000-01-01T00:00+01:23,
+    2000-01-01T12:34:56.789-02:34,
+  ],
+  should_reject_as_invalid:[
+    null,
+    null.timestamp,
+    2000-01-01T00:00+02:34,
+    2000-01-01T00:00Z,
+    2000T,
+    2001-02-03T,
+    2001-02-03T04:05:06-01:23,
+  ]
+}
+
+$test::{
+  description: "timestamp_offset must only be a list of offset strings",
+  invalid_types:[
+    { timestamp_offset: null },
+    { timestamp_offset: null.list },
+    { timestamp_offset: null.string },
+    { timestamp_offset: []},
+    { timestamp_offset: range::["+00:00", "+12:00"] },
+    { timestamp_offset: [foo::"+00:00"] },
+    { timestamp_offset: ("+00:00") },
+    { timestamp_offset: "+00:00" },
+    { timestamp_offset: ["Z"] },
```

Comment:
Seems like it would be fairly easy to allow this... should we?

Reply:
The whole `timestamp_offset` format is weird to begin with—it's a list of strings that we parse to get numbers. Unfortunately, I don't see any clearly better option. Allowing `"Z"` is something we _could_ do, but on the other hand, I suspect that requiring _only_ UTC timestamps is quite common, so perhaps it would be better to allow `timestamp_offset: Z,` as syntactical sugar.  All that to say I'd rather not deal with the `timestamp_offset` format right now—let's wait until we have a compelling reason to make the change.

Reply:
Works for me.